app/src/main/java/org/oppia/android/app/player/state/StatePlayerRecyclerViewAssembler.kt

```diff
@@ -201,7 +204,8 @@ class StatePlayerRecyclerViewAssembler private constructor(
   fun compute(
     ephemeralState: EphemeralState,
     gcsEntityId: String,
-    isSplitView: Boolean
+    isSplitView: Boolean,
+    isTablet: Boolean
```

Comment:
This seems like a fixed parameter, not something that could change between compute calls. Should it be passed into the initialization for the feature or into build() itself?

Also, if possible can we specialize this based on the behavior rather than the configuration in which that behavior occurs? E.g. maybe the differentation should be an enum denoting the celebration type (e.g. streaming vs. burst), and then the calling fragments can decide that one should be associated to a tablet and the other to non-tablet. I think this makes the API a bit more robust. WDYT? In that case, you might be able to associate the value with the feature directly rather than the assembler class or compute method.

Reply:
> This seems like a fixed parameter, not something that could change between compute calls. Should it be passed into the initialization for the feature or into build() itself?
Ah that does make sense, will add to the build() function for the assembler.

> Also, if possible can we specialize this based on the behavior rather than the configuration in which that behavior occurs? E.g. maybe the differentation should be an enum denoting the celebration type (e.g. streaming vs. burst), and then the calling fragments can decide that one should be associated to a tablet and the other to non-tablet. I think this makes the API a bit more robust. WDYT? In that case, you might be able to associate the value with the feature directly rather than the assembler class or compute method.

Reply:
Done
Added tablet Boolean to the full screen confetti feature

Reply:
My concern is more of "should assembler care about what a tablet is?" versus "how should we customize specific features the assembler supports?". In this case, 'tablet' feels a bit like an implementation detail of the host (e.g. the exploration player wants to play a specific kind of confetti animation when the user is using a tablet versus when they aren't). To that end, 'tablet' is the wrong semantic to be used here.

Whether an enum vs. a boolean is used isn't as important, but enums are generally more readable & extensible since they can easily be added onto, and make it very explicit what the intended value means (vs. having to infer meaning from 'true' and 'false').

Reply:
Ah I'm understanding this nuance now, and I see how enums serve as a better representation of how this feature is implemented & customized. If I'm understanding this correctly I think enums would be beneficial if we want to use them to distinguish tablet bursts from mobile bursts, in addition to streaming. Is that a reasonable idea?

My hesitation is that the enums would then store a lot of fields -- this is what I'm imagining if I'm understanding this currectly:

```
enum class ConfettiAnimationType(
  val minSpeed: Float,
  val maxSpeed: Float,
  val particleSizeInDp: Size,
  val particleSizeWithMass: Size,
  val numPieces: Int,
  val timeToLiveMs: Long,
  val delayMs: Long
) {
  CORRECT_ANSWER_BANNER(
    minSpeed = ...,
    maxSpeed = ...,
    ...
  ),
  END_OF_EXPLORATION_MOBILE(...),
  END_OF_EXPLORATION_TABLET(...),
  STREAM(...);
}
```

Reply:
Ah, I wasn't thinking of tying the configuration to the enum in this way, but you certainly can. These would be fixed values that are always true. If you can use the exact same code to spawn the confetti from just these parameters, this is a really nice way to implement it. Alternatively, if the functionality also differs you may want to consider using Kotlin sealed classes, instead. Then you could set something up like:

```kotlin
sealed class ConfettiConfig {
  internal abstract fun showConfetti(...)

  object UpwardBurst: ConfettiConfig() {
    override fun showConfetti(...) {
      ...
    }
  }
}
```

Also one other thing: keep in mind my earlier feedback about being overly prescriptive on where the assembler is used versus what the assembler itself is responsible for. 'tablet' and 'exploration' probably are not the right descriptors to use here. You also may not want to conflate different types of confetti animations together, though that could be another way to design it:

```kotlin
assemblerBuilder
  .setConfettiOnCorrectAnswer(ConfettiConfig.UpwardBurst, forViewId = R.id.target_confetti_view)
  .setConfettiOnSessionCompletion(
    if (isTablet) ConfettiConfig.DownwardScreenBurst else ConfettiConfig.ScreenCornerBurst,
    forViewId = R.id.target_confetti_view
   )
```

(names are placeholders & need more thought)

Reply:
Yeah I think it would be beneficial to provide more configuration details than just a burst or a stream largely since the library does that differentiation for us when we call a burst or stream. I like the idea of the sealed class as we're also able to give it the specific KonfettiView which makes it more versatile for future confetti. I'll try that.

Reply:
Done
I used a sealed class for the configurations and also decided that it made sense to add the burst animation call in the sealed class as well to simplify the usage of the confetti